planb/update_cluster.py

```diff
@@ -358,10 +358,15 @@ def step_forward(ec2: object, volume_id: str, options: dict):
     return True
 
 
-def ssh_command_works(odd_host: str):
+def ssh_command_works(odd_host: str) -> bool:
     ssh = subprocess.Popen(['ssh', odd_host, 'echo', 'test-ssh'])
-    out, err = ssh.communicate()
-    return out == 'test-ssh\n'
+    try:
+        out, err = ssh.communicate(timeout=5)
+        return out == 'test-ssh\n'
+    except Exception as e:
+        logger.error("Failed to open SSH connection to the Odd host: {}".format(e))
+        ssh.kill()
+        ssh.communicate()
```

Comment:
The fine manual says it's nice to do so: https://docs.python.org/3/library/subprocess.html#subprocess.Popen.communicate

I guess it's because of "Wait for process to terminate." in the function description, which doesn't happen if there was a timeout.